### PR TITLE
[IMPROVE] Add support to Linebreak and Katex on markdown

### PR DESCRIPTION
**Pull request by GleidsonDaniel**
## Proposed changes

Fixes the issue of texts in markdown not skipping lines when the component is Code, and the spacing when having a line break.
Add KaTeX support.

## Issue(s)	

Closes https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/1973

## How to test or reproduce

- View any message with spacing or inline code.
- Test KaTeX text.
Send this message to other user: 
```
\[ f(x) = \int_{-\infty}^\infty \hat f(\xi)\,e^{2 \pi i \xi x} \,d\xi \]
```

## Screenshots
Before:
Screen Shot 2022-07-13 at 09 55 05

After:
Screen Shot 2022-07-13 at 09 54 45

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [X] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Inline KaTeX is currently not supported due to parse issues, however there is an open issue:
https://github.com/RocketChat/fuselage/issues/775

### CODE REVIEW

File: app/containers/markdown/new/Katex.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import MathView, { MathText } from 'react-native-math-view';
+
+interface IKaTeXProps {
+	value: KaTeXProps['value'];
+}
+
+export const KaTeX = ({ value }: IKaTeXProps): React.ReactElement | null => <MathView math={value} />;
+
+export const InlineKaTeX = ({ value }: IKaTeXProps): React.ReactElement | null => (
+	<MathText value={`$${value}$$`} direction='ltr' />
```
Comment (diegolmello) [inline app/containers/markdown/new/Katex.tsx:13]: Missing color text to work on dark themes.

Comment (diegolmello) [inline app/containers/markdown/new/Katex.tsx:13]: Is it going to work on `arabic` language, which forces RTL?

File: app/containers/markdown/new/Katex.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { KaTeX as KaTeXProps } from '@rocket.chat/message-parser';
+// eslint-disable-next-line import/no-unresolved
```
Comment (diegolmello) [inline app/containers/markdown/new/Katex.tsx:3]: Is this needed?

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/containers/markdown/new/Katex.tsx (hunk 2 above)
Comment (GleidsonDaniel) [reply app/containers/markdown/new/Katex.tsx:3]: Yep

File: app/containers/markdown/new/Katex.tsx (hunk 1 above)
Comment (GleidsonDaniel) [reply app/containers/markdown/new/Katex.tsx:13]: Mathematics is a language, like javascript, so Arabs or Brazilians speak the same language.
As I remember there are some calculations that can be written in RTL but the default is worldwide.
An edge case so to speak, if someone "complains" we can think about it in the future.

**Pull request merged**